aidial_assistant/chain/history.py

```diff
@@ -32,27 +55,20 @@ def __init__(
         )
         self.best_effort_template = best_effort_template
         self.scoped_messages = scoped_messages
+        self._user_message_count = sum(
+            1
+            for message in scoped_messages
+            if message.scope == MessageScope.USER
+        )
 
     def to_protocol_messages(self) -> list[Message]:
```

Comment:
Let's turn into a private helper `_format_assistant_commands` and rename `to_protocol_messages_with_system_message` back to `to_protocol_messages`.

Reply:
Renamed.